src/hook.c

```diff
@@ -26,8 +26,23 @@
  *  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
+#ifndef _MSC_VER
+#define  _WIN32_WINNT 0x0501
+#define WIN32_LEAN_AND_MEAN
+#endif
 #include <Windows.h>
 #include <TlHelp32.h>
+#ifndef _MSC_VER
+#include <x86intrin.h>
+#include <limits.h>
+#ifndef ARRAYSIZE
+  #define ARRAYSIZE(x) sizeof(x)/sizeof(*x)
```

Comment:
`#define ARRAYSIZE(x) (sizeof(x)/sizeof(*(x)))` - macros without brackets is playing with fire.